### PR TITLE
Fix documentation for printing `ModelMetadata`

### PR DESCRIPTION
**Pull request by frostedoyster**
The documentation at https://docs.metatensor.org/latest/atomistic/reference/models/metadata.html
says there is a `ModelMetadata.print()` method, but it doesn't work at the moment because it's not registered

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [ ] Documentation updated (for new features)?
 - [ ] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [ ] CHANGELOG updated with public API or any other important changes?

----
 📚 [Download documentation preview for this pull-request](https://nightly.link/lab-cosmo/metatensor/actions/artifacts/1672240911.zip)

### CODE REVIEW

**Comment by frostedoyster** [comment]
IDK, works locally

File: metatensor-torch/src/register.cpp [hunk 1]
```diff
@@ -382,6 +382,7 @@ TORCH_LIBRARY(metatensor, m) {
         )
         .def("__repr__", &ModelMetadataHolder::print)
         .def("__str__", &ModelMetadataHolder::print)
+        .def("print", &ModelMetadataHolder::print)
```
Comment (Luthaf) [inline metatensor-torch/src/register.cpp:385]: this is registered as `__str__`; so you can do `print(metadata)` from Python and you don't need to do `print(metadata.print())`

Comment (frostedoyster) [reply metatensor-torch/src/register.cpp:385]: I know, but then the docs are wrong (?)

Comment (frostedoyster) [reply metatensor-torch/src/register.cpp:385]: I just wanted to print, looked up the docs, did `print()`, didn't work. Should we change the Python docs?

Comment (Luthaf) [reply metatensor-torch/src/register.cpp:385]: yes!

**Review by Luthaf — APPROVED** [review]

**Pull request merged**